.github/workflows/release.yml

```diff
@@ -19,79 +30,94 @@ jobs:
+          git add docs/src/antora.yml
+          find . -name 'pom.xml' | xargs git add
+          git commit -m "chore: release version ${{ github.event.inputs.version }}"
 
-      - name: Commit and push update antora.yml timefold-solver
-        uses: actions-js/push@v1.4
-        with:
-          directory: ./timefold-solver
-          github_token: ${{ secrets.JRELEASER_GITHUB_TOKEN }}
-          message: Update antora.yml for release ${{ github.event.inputs.version }}
+      # JReleaser requires the release branch to exist, so we need to push it before releasing.
+      # Once this is pushed, branch protection rules apply.
+      # So if any of the subsequent steps should fail, the release branch is there to stay; cannot be deleted.
+      # To minimize that chance, do a dry run first, with a branch named in a way that the protection rules don't apply.
+      - name: Push release branch to Git
+        run: |
+          git push origin ${{ github.event.inputs.releaseBranch }}
 
-      - name: Run JReleaser timefold-solver
-        working-directory: ./timefold-solver
-        run: mvn --no-transfer-progress --batch-mode -Dfull -pl . jreleaser:full-release
+      - name: Run JReleaser
+        run: mvn --no-transfer-progress --batch-mode -Dfull -Djreleaser.dry.run=${{ github.event.inputs.dryRun }} -pl . jreleaser:full-release
         env:
           JRELEASER_GITHUB_TOKEN: ${{ secrets.JRELEASER_GITHUB_TOKEN }}
           JRELEASER_GPG_PASSPHRASE: ${{ secrets.JRELEASER_GPG_PASSPHRASE }}
           JRELEASER_GPG_PUBLIC_KEY: ${{ secrets.JRELEASER_GPG_PUBLIC_KEY }}
           JRELEASER_GPG_SECRET_KEY: ${{ secrets.JRELEASER_GPG_SECRET_KEY }}
-          JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME: ${{ secrets.JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME }}
-          JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD: ${{ secrets.JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD }}
+          JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME: ${{ secrets.JRELEASER_NEXUS2_USERNAME }}
+          JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD: ${{ secrets.JRELEASER_NEXUS2_PASSWORD }}
 
-      - name: JReleaser release output timefold-solver
-        if: always()
+      - name: JReleaser release output
         uses: actions/upload-artifact@v3
+        if: always()
         with:
           name: jreleaser-release
           path: |
-            timefold-solver/target/jreleaser/trace.log
-            timefold-solver/target/jreleaser/output.properties
+            target/jreleaser/trace.log
+            target/jreleaser/output.properties
 
-      - name: Set snapshot version timefold-solver
-        working-directory: ./timefold-solver
-        run: mvn --no-transfer-progress --batch-mode -Dfull versions:set -DnewVersion=${{ github.event.inputs.nextVersion }}-SNAPSHOT
+      - name: Set micro snapshot version on the release branch
+        run: |
+          mvn --no-transfer-progress --batch-mode -Dfull versions:set -DnewVersion=${{ github.event.inputs.nextMicroVersion }}-SNAPSHOT
```

Comment:
Something for consideration: define some MAVEN_OPTIONS (?) env var to keep the `-no-transfer-progress --batch-mode` in a single place.